src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -35,6 +35,9 @@ namespace Microsoft.CodeAnalysis.CodeActions
     /// </summary>
     public abstract class CodeAction
     {
+        private static readonly ConditionalWeakTable<Type, StrongBox<bool>> s_isNonProgressGetChangedSolutionAsyncOverridden = new();
+        private static readonly ConditionalWeakTable<Type, StrongBox<bool>> s_isNonProgressComputeOperationsAsyncOverridden = new();
```

Comment:
💡 `ConditionalWeakTable` is unnecessary overhead here. The `Type` key is always going to be rooted outside this collection, so you'd be better off making this a normal dictionary

Reply:
sgtm.

Reply:
done.